### ISSUE TITLE
MA0112 shouldn't trigger on `IEnumerable<T>`

### ISSUE THREAD

**Issue by Sella-GH** [meziantou/Meziantou.Analyzer#1303]
### Version of the Meziantou.Analyzer NuGet package

3.0.167

### Rule Identifier

MA0112

### Target Framework

net10.0

### C# Language version

C# 14

### .NET SDK version

10.0.400

### Description

`IEnumerable<T>` does not have a `Count` property, so the Analyzer is misleading.
Also the proposed Code Fix does not work as the type doesn't have the property, resulting in a compiler error. Changing it to the correct method group `Count()` triggers CA1827.

### Reproduction Steps

I can't get a minimal reproducible as it strangely works whenever I do it regularly. However [this code](https://github.com/Sella-GH/AzzyBot/blob/343f41026d36bae91201b740b8d309a857c036e6/src/AzzyBot.Bot/Services/Modules/AzuraCastApiService.cs?plain=1#L119-L121) does trigger the Analyzer.

```c#
IEnumerable<string> missing = await ExecuteApiPermissionCheckAsync(apis, apiKey);
if (!missing.Any()) // Trigger
    return;
```
